### PR TITLE
Documentation: Add performance-analysis.md about manual analyzer profiling

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
While working on #6964, I used the VS performance tools and the Analyzer Runner to find allocation hotspots (see e.g. #7012 for how this looked in practice). This PR documents how this was done so we can repeat it in the future. We can also use the same approach and integrate the performance diagnostics in the CI because there is a CLI that can be used to capture performance traces.

### CODE REVIEW

File: docs/perfomance-analysis.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Analyze a rules performance with the [Visual Studio performance profiler](https://learn.microsoft.com/en-us/visualstudio/profiling)
```
Comment (pavel-mikula-sonarsource) [inline docs/perfomance-analysis.md:1]: I wouldn't put this in the repo itself. Users don't need to do this and it's hard to maintain and update.

Make it an xtranet page or google doc in the shared drive. I'd propose to sync with @costin-zaharia-sonarsource as he might have an idea how to organize some "Performance" folder there.

Comment (costin-zaharia-sonarsource) [reply docs/perfomance-analysis.md:1]: I would move this to xtranet. 

Additionally, this is not enough, this approach is very useful to isolate local problems but it's not precise enough for normal execution during build. For sure we can start with this and then extend for measuring the full build.

Comment (martin-strecker-sonarsource) [reply docs/perfomance-analysis.md:1]: > For sure we can start with this and then extend for measuring the full build.

This is for tracking down problems and not so much for identifying them.

Comment (costin-zaharia-sonarsource) [reply docs/perfomance-analysis.md:1]: We do have this tips and tricks page on xtranet. You can add it there: https://xtranet-sonarsource.atlassian.net/wiki/spaces/LANG/pages/2690089021/Tips+Tricks+-+.NET+Squad

Comment (martin-strecker-sonarsource) [reply docs/perfomance-analysis.md:1]: I copied the content to https://xtranet-sonarsource.atlassian.net/wiki/spaces/LANG/pages/2788524068/Analyze+a+rules+performance+with+the+Visual+Studio+performance+profiler

**Comment by martin-strecker-sonarsource** [comment]
Content moved to https://xtranet-sonarsource.atlassian.net/wiki/spaces/LANG/pages/2788524068/Analyze+a+rules+performance+with+the+Visual+Studio+performance+profiler

**Pull request closed without merging**